http/src/main/scala/org/http4s/blaze/http/http2/OutboundStreamState.scala

```diff
@@ -0,0 +1,62 @@
+package org.http4s.blaze.http.http2
+
+import scala.concurrent.Promise
+
+/** Representation of outbound streams
+  *
+  * We need a concrete StreamState to send messages to, but we can't
+  * expect that we will have HEADERS to send right when it is born, so
+  * we need to make the stream ID lazy since they must be used in
+  * monotonically increasing order.
+  */
+private final class OutboundStreamState(session: SessionCore) extends StreamState(session) {
+  private[this] var lazyStreamId: Int = -1
+  private[this] var lazyFlowWindow: StreamFlowWindow = null
+
+  private[this] def initialized: Boolean = lazyStreamId != -1
+  private[this] def uninitializedException: Nothing = {
```

Comment:
Unimportant because it's private, but without parens on this, I was expecting to find a cached exception until I scrolled back up.

Reply:
I can add them.